tests/unit/other.tcl

```diff
@@ -471,4 +484,48 @@ start_cluster 1 0 {tags {"other external:skip cluster slow"}} {
         after 200;# waiting for serverCron
         assert_match "*table size: 8*" [r debug HTSTATS 0]
     } {} {needs:debug}
+
+    # Tune up hz to speed up cron
+    set default_hz [lindex [r config get hz] 1]
+    r config set hz 200
+
+    # The overhead_hashtable_main is calculated by:
+    # mem = keys_count * (sizeof(robj) + dictEntryMemUsage()) +
+    #       db->dict_count * sizeof(dict) +
+    #       dbBuckets(db, keyType) * sizeof(dictEntry*);
+    # and we want to compare the 3rd line.
+    test "Redis can resize empty dict" {
+        # Get the value of sizeof(robj) and dictEntryMemUsage()
+        set res [r debug structsize]
+        set items [split $res " "]
+        foreach item $items {
+            set kv [split $item ":"]
+            set name [lindex $kv 0]
+            set value [lindex $kv 1]
+            if {[string equal -nocase "robj" $name]} {
+                set robj $value
+            } elseif {[string equal -nocase "dictentry" $name]} {
+                set dictentry $value
+            }
+        }
+        # Write and then delete 128 keys, creating an empty dict
+        r flushall
+        for {set j 1} {$j <= 128} {incr j} {
+            r set $j{b} a
+        }
+        set dict1 [get_overhead_hashtable_main]
+        for {set j 1} {$j <= 128} {incr j} {
+            r del $j{b}
+        }
+        # Set a key to enable overhead display of db 0
+        r set a b
+        # Compare to make sure the dict is resized
+        wait_for_condition 100 50 {
+            [expr $dict1-($robj+$dictentry)*127] > [get_overhead_hashtable_main]
```

Comment:
i agree. this code is far to specific.
i rather avoid using DEBUG STRUCTSIZE, and instead just set some threshold like 200 bytes or alike.

Reply:
To make sure a resizing of a dict actually happened, we need to know exactly how many bytes are reduced due to the deletion of keys. An approxiamte threshold may give wrong results on different platforms.

Reply:
This difference may be caused by the deletion of keys rather than the resizing, see `size_t dbMemUsage(redisDb *db, dbKeyType keyType)`. I wrote those calculaiton in the test just to make sure: other than the memory reduced by the deletion of keys, there are still memory reduced by resizing.

 I wonder how we can obtain a proper threshold.

Reply:
but still the difference between a dict with an allocated lookup table and one without is huge.
i.e. just `struct dict` is about 56 bytes, and with the lookup table for 128 keys, it's over 1kb (even without dict entries).
so checking that it's less than 100 bytes, should be sufficient for this test.
comment can say that we check that the dict struct is allocated, but the hash table itself is not. (without any specific calculations)

Reply:
I was think of running this test in cluster mode. I just realized it is still effective in standalone mode.